tests/meshcop/commissioner_argcargv.cpp

```diff
@@ -354,100 +336,108 @@ static void handle_pskd(argcargv *pThis)
 {
-    pThis->str_param(gContext.mAgent.mPassPhrase, sizeof(gContext.mAgent.mPassPhrase));
+    pThis->StrParam(args->mPassPhrase, sizeof(args->mPassPhrase));
 }
 
 /** Handle log fileanme on the command line */
-static void handle_log_filename(argcargv *pThis)
+static void handle_log_filename(ArgcArgv *pThis, CommissionerArgs *args)
 {
+    (void)args;
     char filename[PATH_MAX];
 
-    pThis->str_param(filename, sizeof(filename));
+    pThis->StrParam(filename, sizeof(filename));
 
     otbrLogSetFilename(filename);
 }
 
 /** compute the pskc from command line params */
-static void handle_compute_pskc(argcargv *pThis)
+static void handle_compute_pskc(ArgcArgv *pThis, CommissionerArgs *args)
 {
-    (void)(pThis);
-    CommissionerComputePskc();
+    (void)pThis;
+    args->mNeedComputePSKc = true;
     /* we print this in a way scripts can easily parse */
-    fprintf(stdout, "PSKc: %s\n", gContext.mAgent.mPSKc.ascii);
-    exit(EXIT_SUCCESS);
+    // fprintf(stdout, "PSKc: %s\n", gContext.mAgent.mPSKc.ascii);
+    // exit(EXIT_SUCCESS);
 }
 
 /** commandline handling for flag that says we commission (not test) */
-static void handle_commission_device(argcargv *pThis)
+static void handle_commission_device(ArgcArgv *pThis, CommissionerArgs *args)
 {
-    (void)(pThis);
-    gContext.commission_device = true;
+    (void)pThis;
+    args->mNeedCommissionDevice = true;
 }
 
 /** compute hashmac of EUI64 on command line */
-static void handle_compute_hashmac(argcargv *pThis)
+static void handle_compute_hashmac(ArgcArgv *pThis, CommissionerArgs *args)
 {
     (void)pThis;
+    args->mNeedComputeJoinerHashMac = true;
 
-    CommissionerComputeHashMac();
     /* print so scripts can easily parse */
-    fprintf(stdout, "eiu64: %s\n", gContext.mJoiner.mEui64.ascii);
-    fprintf(stdout, "hashmac: %s\n", gContext.mJoiner.mHashMac.ascii);
-    exit(EXIT_SUCCESS);
+    // fprintf(stdout, "eiu64: %s\n", gContext.mJoiner.mEui64.ascii);
+    // fprintf(stdout, "hashmac: %s\n", gContext.mJoiner.mHashMac.ascii);
+    // exit(EXIT_SUCCESS);
 }
 
 /** compute steering based on command line */
-static void handle_compute_steering(argcargv *pThis)
+static void handle_compute_steering(ArgcArgv *pThis, CommissionerArgs *args)
 {
     (void)pThis;
+    args->mNeedComputeJoinerSteering = true;
 
-    CommissionerComputeSteering();
+    // CommissionerComputeSteering();
 
-    /* print so scripts can easily parse */
-    fprintf(stdout, "eiu64: %s\n", gContext.mJoiner.mEui64.ascii);
-    fprintf(stdout, "hashmac: %s\n", gContext.mJoiner.mHashMac.ascii);
-    fprintf(stdout, "steering-len: %d\n", gContext.mJoiner.mSteeringData.GetLength());
-    fprintf(stdout, "steering-hex: %s\n",
-            CommissionerUtilsHexString(gContext.mJoiner.mSteeringData.GetDataPointer(),
-                                       gContext.mJoiner.mSteeringData.GetLength()));
-
-    exit(EXIT_SUCCESS);
+    ///* print so scripts can easily parse */
```

Comment:
Delete this code?

Reply:
The test script now depends on this print to get correct pskc